javalib/src/main/scala/java/lang/String.scala

```diff
@@ -633,18 +633,52 @@ final class _String()
     buffer
   }
 
-  def toLowerCase(locale: Locale): _String = ???
+  def toLowerCase(locale: Locale): _String = {
+    toCase(locale, Character.toLowerCase)
+  }
 
   def toLowerCase(): _String =
     toLowerCase(Locale.getDefault)
 
   override def toString(): String = this
 
-  def toUpperCase(locale: Locale): _String = ???
-
+  def toUpperCase(locale: Locale): _String = {
+    toCase(locale, Character.toUpperCase)
+  }
   def toUpperCase(): _String =
     toUpperCase(Locale.getDefault)
 
+  private[this] def toCase(locale: Locale, convert: Int => Int): _String = {
+    if (count == 0) value
```

Comment:
Did you mean `if (count == 0) return value` ? Currently this check doesn't do anything.